service/src/main/java/bio/terra/drshub/util/AccessMethodUtils.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.drshub.util;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+
+import bio.terra.drshub.config.DrsProvider;
+import bio.terra.drshub.generated.model.RequestObject.CloudPlatformEnum;
+import io.github.ga4gh.drs.model.AccessMethod;
+import io.github.ga4gh.drs.model.DrsObject;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class AccessMethodUtils {
+  public static Optional<AccessMethod> getAccessMethod(
+      DrsObject drsResponse, DrsProvider drsProvider, CloudPlatformEnum cloudPlatform) {
+    Optional<AccessMethod> accessMethod = Optional.empty();
+    if (!isEmpty(drsResponse)) {
+      List<AccessMethod> accessMethods = getAccessMethods(drsResponse, drsProvider);
+      if (cloudPlatform != null) {
+        accessMethod = getAccessMethodForCloud(accessMethods, cloudPlatform);
+      }
+      // if there is no access method matching the cloudPlatform, or
+      // if the cloudPlatform was not specified, return a different access method
+      if (accessMethod.isEmpty() && !accessMethods.isEmpty()) {
+        accessMethod = Optional.of(accessMethods.get(0));
```

Comment:
To be consistent with below, I would do:
```suggestion
        accessMethod = accessMethods.stream().findFirst();
```

Reply:
And if you use that syntax, then you don't even need the `&& !accessMethods.isEmpty()` above.

You could actually just write the if + return part as:
```return accessMethod.orElseGet(() -> accessMethods.stream().findFirst())```